core/src/main/java/org/bitcoinj/crypto/bls/BLSDeterministicKey.java

```diff
@@ -96,10 +94,12 @@ public BLSDeterministicKey(ImmutableList<ChildNumber> childNumberPath,
             bytes = serialize(null, false, Script.ScriptType.P2PKH);
             extendedPrivateKey = ExtendedPrivateKey.fromBytes(bytes);
             extendedPublicKey = extendedPrivateKey.getExtendedPublicKey();
+            this.parentFingerprint = (parent != null) ? (int)extendedPrivateKey.getParentFingerprint() : 0;
         } else {
             bytes = serialize(null, true, Script.ScriptType.P2PKH);
             extendedPrivateKey = null;
             extendedPublicKey = ExtendedPublicKey.fromBytes(bytes);
+            this.parentFingerprint = (parent != null) ? (int)extendedPublicKey.getParentFingerprint() : 0;
```

Comment:
These two lines look identical and seem like they override the earlier assignment:
`this.parentFingerprint = (parent != null) ? parent.getFingerprint() : 0;`

Perhaps this can be simplified by leaving one assigment after `if/else`

Reply:
Thanks.  Will do that in the next PR.